### PR TITLE
DEP: Execute deprecation of pinv2

### LINKED ISSUE #15737: DEP: Execute deprecation of pinv2
From #13831. That PR does not specify a specific removal date, so I think we could sharpen the warnings to indicate removal by 1.10/1.11; on the other hands, two releases between deprecation and removal is also common practice, so would be an option as well IMO. In either case, there's something actionable for 1.9, hence the milestone.

CC @ilayn

### PR DESCRIPTION
**Pull request by j-bowhay**
#### Reference issue

Closes #15737
#### What does this implement/fix?

pinv2 has been deprecated since 1.7.0 so could be removed for 1.9.0

### CODE REVIEW

File: scipy/linalg/tests/test_basic.py [hunk 1]
```diff
@@ -1277,48 +1277,34 @@ def test_simple_real(self):
...
         a_pinv = pinv(a)
         assert_array_almost_equal(dot(a, a_pinv), np.eye(3))
-        a_pinv = pinv2(a)
-        assert_array_almost_equal(dot(a, a_pinv), np.eye(3))
 
     def test_simple_singular(self):
         a = array([[1, 2, 3], [4, 5, 6], [7, 8, 9]], dtype=float)
         a_pinv = pinv(a)
-        a_pinv2 = pinv2(a)
-        assert_array_almost_equal(a_pinv, a_pinv2)
```
Comment (ilayn) [inline scipy/linalg/tests/test_basic.py:1296]: When these lines removed the test loses its assertions, hence needs to be replaced by the actual pinv result.

File: scipy/linalg/tests/test_basic.py [hunk 2]
```diff
@@ -1277,48 +1277,34 @@ def test_simple_real(self):
...
 
     def test_simple_cols(self):
         a = array([[1, 2, 3], [4, 5, 6]], dtype=float)
         a_pinv = pinv(a)
-        a_pinv2 = pinv2(a)
-        assert_array_almost_equal(a_pinv, a_pinv2)
```
Comment (ilayn) [inline scipy/linalg/tests/test_basic.py:1302]: Same as above

File: scipy/linalg/tests/test_basic.py [hunk 3]
```diff
@@ -1277,48 +1277,34 @@ def test_simple_real(self):
...
 
     def test_simple_rows(self):
         a = array([[1, 2], [3, 4], [5, 6]], dtype=float)
         a_pinv = pinv(a)
-        a_pinv2 = pinv2(a)
-        assert_array_almost_equal(a_pinv, a_pinv2)
```
Comment (ilayn) [inline scipy/linalg/tests/test_basic.py:1308]: Same as above

File: scipy/linalg/tests/test_basic.py [hunk 4]
```diff
@@ -1277,48 +1277,34 @@ def test_simple_real(self):
...
         a_pinv = pinv(a, check_finite=False)
         assert_array_almost_equal(dot(a, a_pinv), np.eye(3))
-        a_pinv = pinv2(a, check_finite=False)
-        assert_array_almost_equal(dot(a, a_pinv), np.eye(3))
 
     def test_native_list_argument(self):
         a = [[1, 2, 3], [4, 5, 6], [7, 8, 9]]
         a_pinv = pinv(a)
-        a_pinv2 = pinv2(a)
-        assert_array_almost_equal(a_pinv, a_pinv2)
```
Comment (ilayn) [inline scipy/linalg/tests/test_basic.py:1321]: Same as above

**Review by ilayn — CHANGES_REQUESTED** [review]
Thanks @j-bowhay, commented on some minor issue.

**Comment by ilayn** [comment]
I am so **not** looking forward to this about 😃 But better do it now than later. Thanks @j-bowhay  Do you mind adding a line to the wiki about it?

**Comment by j-bowhay** [comment]
> Do you mind adding a line to the wiki about it?

yep will do

**Comment by WarrenWeckesser** [comment]
There are a couple references to `pinv2` in the linalg tutorial that should be cleaned up:

https://github.com/scipy/scipy/blob/62f75fa33f1a832043b53f0694693dee0fe9503a/doc/source/tutorial/linalg.rst?plain=1#L354-L359

https://github.com/scipy/scipy/blob/62f75fa33f1a832043b53f0694693dee0fe9503a/doc/source/tutorial/linalg.rst?plain=1#L404-L409

**Comment by ilayn** [comment]
Argh. Missed those. Thanks for the heads up Warren.

**Pull request merged**